RELEASES.md

```diff
@@ -0,0 +1,34 @@
+# Releases
+
+## Overview
+
+Releases on this project are triggered through GitHub Actions listening for differences between our `main` and `develop` branches. A pull request for the next release is created by a github-actions bot and merging this pull request automatically delivers the changes up to production.
```

Comment:
@travissouthard once we finish, the setup _won't_ be that changes get delivered up to production. The result will be a new version is available, and then it's a separate PR against Code for Philly's live cluster to increase what version Code for Philly has deployed. 

There is some interest in trying out what it would be like to automate that, we could give it a shot as a next step